docs/design/mojaloop_integration.md

```diff
@@ -0,0 +1,160 @@
+Mojaloop's RTP network.
+
+![Figure 3](http://www.plantuml.com/plantuml/proxy?cache=no&src=https://raw.githubusercontent.com/stevenwjy/pisp-demo-server/mojaloop-integration-docs/docs/assets/diagrams/mojaloop/fig-03-pisp-mojaloop.puml)
+
+In order to join a Mojaloop network, a PISP must complete the following steps:
+1. [Certificate Registration](#certificate-registration)
+2. [URL Registration](#url-registration)
+
+In general, all of the signing, encryption, and other setup required to make requests to Mojaloop 
+are expected to be handled by the [`sdk-standard-components`](https://github.com/mojaloop/sdk-standard-components)
+libary. Hence, PISP demo server only needs to set the proper configuration such as path to the 
+issued certificate, Mojaloop's URL, PISP identifier, transport scheme, and various other options
+which could be found [here](https://github.com/mojaloop/sdk-standard-components/blob/master/src/lib/requests/baseRequests.js).
+
+## Certificate Registration
+
+In the Mojaloop's network for FSP Interoperability, every participant must be registered and have a
+signed certificate by the centralized Certificate Authority (CA). According to the 
+[PKI Best Practices](https://docs.mojaloop.io/mojaloop-specification/documents/PKI%20Best%20Practices.html),
+the CA itself will have a self-signed root certificate for signing the participants' certificates. 
+The certificates are important to protect the transport-level communication (TLS) and application-level
+communication (JWS and JWE). All of the security protocols involved are for the purpose of ensuring
+integrity and confidentiality between platforms. More information could be found in the 
+[Mojaloop Specifications](https://docs.mojaloop.io/mojaloop-specification/).
+
+PISP has to register its certificate manually, by telling the operator of Mojaloop the PISP's public
+key and have it digitally signed by the centrallized CA.
+
+## URL Registration
+
+As Mojaloop's architecture follows the asynchronous request/response design pattern, PISP demo server 
+also interacts with Mojaloop using that pattern to avoid long-running connections. When making a 
+request to Mojaloop, it may take some time to receive the response as everything is considered to work 
+asynchronously. Even though it takes only several seconds on average, we still need to prepare for the 
+indefinite time characteristic of asynchronous messages. As a result, PISP demo server needs to register 
+its URL so that Mojaloop knows where to make the callback once the reply is ready. For example, 
+Mojaloop needs to know the endpoint of the PISP server that is ready to handle `PUT /parties/{Type}/{ID}` 
+callback from Mojaloop following a previous request of `GET /parties/{Type}/{ID}` to perform a party lookup.
+
+The following are the callback URLs that need to be registered by the PISP demo server:
+- `PUT /participants`
+- `PUT /parties/{Type}/{ID}`
+- `PUT /consentRequests/{ID}`
+- `POST /consents`
```

Comment:
I might just be confused about functionality here but wouldn't we also need to mention the specific routes beyond the `.../{ID}/..` such as `POST consent/{ID}/revoke` or `POST consent/{ID}/generateChallenge` ?

Reply:
I think that's unnecessary here because those are the requests that we need to make to Mojaloop; meanwhile, the endpoints that we write down here are the callbacks that we expect Mojaloop to make to the PISP server. 

However, just realized that we haven't included the callback for consent revocation. Will update it